cmd/kubeadm/app/preflight/checks.go

```diff
@@ -920,7 +921,7 @@ func RunInitNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfigura
 
 		// Check if Bridge-netfilter and IPv6 relevant flags are set
 		if ip := net.ParseIP(cfg.LocalAPIEndpoint.AdvertiseAddress); ip != nil {
-			if ip.To4() == nil && ip.To16() != nil {
+			if utilsnet.IsIPv6(ip.To4()) {
```

Comment:
note: while these are not technically equivilant, we call `ParseIP` above which returns nil if the IP is invalid, and `ip.To16() != nil` only accomplishes the same thing as the `ip != nil` above.

same thing for the check below on line 990